src/fetch.rs

```diff
@@ -317,3 +210,84 @@ fn get_cargo_toml_from_git_url(url: &str) -> Result<String> {
         .chain_err(|| "Git response not a valid `String`")?;
     Ok(body)
 }
+
+/// Generate all similar crate names
+///
+/// Examples:
+///
+/// | input | output |
+/// | ----- | ------ |
+/// | cargo | cargo  |
+/// | cargo-edit | cargo-edit, cargo_edit |
+/// | parking_lot_core | parking_lot_core, parking_lot-core, parking-lot_core, parking-lot-core |
+fn gen_fuzzy_crate_names(crate_name: String) -> Result<Vec<String>> {
+    const PATTERN: [u8; 2] = [b'-', b'_'];
+
+    let wildcard_indexs = crate_name
+        .bytes()
+        .enumerate()
+        .filter(|(_, item)| PATTERN.contains(item))
+        .map(|(index, _)| index)
+        .collect::<Vec<usize>>();
+    if wildcard_indexs.len() > 127 {
```

Comment:
actually, even 2**127 is a very big number, maybe we should limit to e.g. 2**16?